cli/run_util.go

```diff
@@ -82,16 +82,39 @@ func (r WorkflowRunner) activateAndRunSteps(
 	runResultCollector := newBuildRunResultCollector(r.logger, tracker)
 	currentStepGroupID := ""
 
+	// Global variables for restricting Step Bundle's environment variables for the given Step Bundle
+	currentStepBundleUUID := ""
+	// TODO: add the last step bundle's envs to environments
+	var currentStepBundleEnvVars []envmanModels.EnvironmentItemModel
+
 	// ------------------------------------------
 	// Main - Preparing & running the steps
 	for idx, stepPlan := range plan.Steps {
-		if stepPlan.GroupID != currentStepGroupID {
-			if stepPlan.GroupID != "" {
+		if stepPlan.WithGroupUUID != currentStepGroupID {
+			if stepPlan.WithGroupUUID != "" {
 				if len(stepPlan.ContainerID) > 0 || len(stepPlan.ServiceIDs) > 0 {
-					r.startContainersForStepGroup(stepPlan.ContainerID, stepPlan.ServiceIDs, *environments, stepPlan.GroupID, plan.WorkflowTitle)
+					r.startContainersForStepGroup(stepPlan.ContainerID, stepPlan.ServiceIDs, *environments, stepPlan.WithGroupUUID, plan.WorkflowTitle)
 				}
 			}
-			currentStepGroupID = stepPlan.GroupID
+
+			currentStepGroupID = stepPlan.WithGroupUUID
+		}
+
+		buildEnvironments := append([]envmanModels.EnvironmentItemModel{}, *environments...)
```

Comment:
Maybe I miss something, but I don't get why it's named `buildEnvironments`

Reply:
Probably the variable name is not the best.

`environments` is the list of env vars that store all the env vars of the given workflow run, which might execute multiple workflows, this is where the `buildEnvironments` name comes from.

For Step Bundles, I had to branch the build env var list, because a Step Bundle can have their own input env vars, and those shouldn't be exposed outside of the Step Bundle.

So when the CLI runs a Step Bundle we separate the builds env var list and the Step Bundle's env var list.
We add the Step Bundle's input env vars to the build env var list (before running the first step in the bundle), this env var list is the `currentStepBundleEnvVars`. Each step's output within the Step Bundle gets added both to the build env var list and to the  `currentStepBundleEnvVars` too.